runtime/common/ArgumentConversion.cpp

```diff
@@ -100,6 +92,60 @@ static Value genConstant(OpBuilder &, cudaq::cc::StructType, void *,
 static Value genConstant(OpBuilder &, cudaq::cc::ArrayType, void *,
                          ModuleOp substMod, llvm::DataLayout &);
 
+static Value genConstant(OpBuilder &builder, const cudaq::state *v,
+                         ModuleOp substMod, llvm::DataLayout &layout,
+                         bool isSimulator) {
+  if (isSimulator) {
+    // The program is executed remotely, materialize the simulation data
+    // into an array and create a new state from it.
+    auto ctx = builder.getContext();
+    auto loc = builder.getUnknownLoc();
+    auto size = 1ULL << v->get_num_qubits();
```

Comment:
Could we please range check `v->get_num_qubits()`? I've seen too many cases where the `<<` operand is >= 64 and then stuff silently goes haywire. (It's especially concerning in this case because `v` could point to invalid data in some erroneous cases.) It would be better to get a clear error message here rather than snowballing into something bigger.

Reply:
Done, thanks!